From 891f8388ddbb51e6ca31252e67145bded45ec841 Mon Sep 17 00:00:00 2001
From: Weiwei Wang <weiwei.wang@windriver.com>
Date: Tue, 8 Dec 2009 00:47:14 -0800
Subject: [PATCH] x86/supermicro: enable COM3 and COM4

Assign COM3 and COM4 to ttyS2 and ttyS3, and switch to irq
mode. Note that the IRQ number defaults align with the
hypervisor xml setup. If you change the xml, you must update
this file.

Signed-off-by: Weiwei Wang <weiwei.wang@windriver.com>
---
 drivers/serial/8250.c         |    3 +++
 include/asm-x86/serial.h      |    7 +------
 include/asm-x86/wrhv_serial.h |   14 ++++++++++++++
 3 files changed, 18 insertions(+), 6 deletions(-)
 create mode 100644 include/asm-x86/wrhv_serial.h

diff --git a/drivers/serial/8250.c b/drivers/serial/8250.c
index 1335654..92025bb 100644
--- a/drivers/serial/8250.c
+++ b/drivers/serial/8250.c
@@ -91,6 +91,9 @@ static unsigned int nr_uarts = CONFIG_SERIAL_8250_RUNTIME_UARTS;
 #define CONFIG_HUB6 1
 
 #include <asm/serial.h>
+#if defined(CONFIG_WRHV) && defined(CONFIG_X86)
+#include <asm/wrhv_serial.h>
+#endif
 /*
  * SERIAL_PORT_DFNS tells us about built-in ports that have no
  * standard enumeration mechanism.   Platforms that can find all
diff --git a/include/asm-x86/serial.h b/include/asm-x86/serial.h
index b6f4ce8..628c801 100644
--- a/include/asm-x86/serial.h
+++ b/include/asm-x86/serial.h
@@ -24,11 +24,6 @@
 	{ 0, BASE_BAUD, 0x3F8, 4, STD_COM_FLAGS },	/* ttyS0 */	\
 	{ 0, BASE_BAUD, 0x2F8, 3, STD_COM_FLAGS },	/* ttyS1 */	\
 	{ 0, BASE_BAUD, 0x3E8, 4, STD_COM_FLAGS },	/* ttyS2 */	\
-	{ 0, BASE_BAUD, 0x2E8, 3, STD_COM4_FLAGS },	/* ttyS3 */	\
-	{ 0, BASE_BAUD, 0x220, 0, STD_COM_FLAGS },      /* ttyS4 */     \
-	{ 0, BASE_BAUD, 0x238, 0, STD_COM_FLAGS },      /* ttyS5 */     \
-	{ 0, BASE_BAUD, 0xDCC8, 0, STD_COM_FLAGS },     /* ttyS6 */     \
-	{ 0, BASE_BAUD, 0xDCD0, 0, STD_COM_FLAGS },     /* ttyS7 */
-
+	{ 0, BASE_BAUD, 0x2E8, 3, STD_COM4_FLAGS },	/* ttyS3 */
 
 #endif /* _ASM_X86_SERIAL_H */
diff --git a/include/asm-x86/wrhv_serial.h b/include/asm-x86/wrhv_serial.h
new file mode 100644
index 0000000..acb3bad
--- /dev/null
+++ b/include/asm-x86/wrhv_serial.h
@@ -0,0 +1,14 @@
+#ifndef _ASM_X86_WRHV_SERIAL_H
+#define _ASM_X86_WRHV_SERIAL_H
+
+#include <asm/serial.h>
+
+#undef SERIAL_PORT_DFNS
+#define SERIAL_PORT_DFNS			\
+	/* UART CLK   PORT IRQ     FLAGS        */			\
+	{ 0, BASE_BAUD, 0x3F8, 4,  STD_COM_FLAGS },	/* ttyS0 */	\
+	{ 0, BASE_BAUD, 0x2F8, 3,  STD_COM_FLAGS },	/* ttyS1 */	\
+	{ 0, BASE_BAUD, 0x220, 30, STD_COM_FLAGS },	/* ttyS2 */	\
+	{ 0, BASE_BAUD, 0x238, 31, STD_COM_FLAGS },	/* ttyS3 */
+
+#endif /* _ASM_X86_WRHV_SERIAL_H */
-- 
1.6.5.2

